pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,569 @@
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, testSchemaName, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": 5},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:44: executing \"reportQueryTemplate\" at <.Report.Inputs.Unrea...>: wrong type for value; expected string; got int",
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to Report.Inputs referencing a *time.Time object) results in error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, testSchemaName, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": reportStart},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:44: executing \"reportQueryTemplate\" at <.Report.Inputs.Unrea...>: wrong type for value; expected string; got *time.Time",
+		},
+		{
+			name: "valid report query with valid templating but PrestoTable contains empty schema field returns error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, "", testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ds1.Name},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: PrestoTable status is invalid: prestoTable.Status.Schema is unset",
+		},
+		{
+			name: "Report dependency was not found (due to no reports) returns error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| reportTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
```

Comment:
This function should be refactored probably to take inputs, it doesn't need `ds1`

Reply:
Also, all of these on reports need a `type: Report` input definition